opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanBuilderShimTest.java

```diff
@@ -433,4 +468,35 @@ public SamplingResult shouldSample(
       return SamplingResult.recordAndSample();
     }
   }
+
+  static final class DebugSampler implements Sampler {
+    static final DebugSampler INSTANCE = new DebugSampler();
+    static final String DEBUG_ID_TAG = "jaeger-debug-id";
+
+    @Override
+    public String getDescription() {
+      return "DebugSampler";
+    }
+
+    @Override
+    public SamplingResult shouldSample(
+        Context parentContext,
+        String traceId,
+        String name,
+        SpanKind spanKind,
+        Attributes attributes,
+        List<LinkData> parentLinks) {
+      SpanContext parentSpanContext = Span.fromContext(parentContext).getSpanContext();
+      if (parentSpanContext.isSampled()
```

Comment:
What's the purpose of this check? Shouldn't the presence of DEBUG_ID_TAG be enough to cause sampling?

Reply:
Because the my current implementation in`Propagation.java` (shim) is to check `span.getSpanContext().isSampled()` in order to return a non-null  `SpanContextShim` for jaeger-debug-id case.

So I just added `parentSpanContext.isSampled()` for one more check here in the test sampler.